pkg/logql/syntax/lex_test.go

```diff
@@ -95,6 +95,19 @@ func TestLex(t *testing.T) {
 		{`{foo="bar"} | logfmt --strict code"`, []int{OPEN_BRACE, IDENTIFIER, EQ, STRING, CLOSE_BRACE, PIPE, LOGFMT, PARSER_FLAG, IDENTIFIER}},
 		{`{foo="bar"} | logfmt --keep-empty --strict code="response.code", IPAddress="host"`, []int{OPEN_BRACE, IDENTIFIER, EQ, STRING, CLOSE_BRACE, PIPE, LOGFMT, PARSER_FLAG, PARSER_FLAG, IDENTIFIER, EQ, STRING, COMMA, IDENTIFIER, EQ, STRING}},
 		{`decolorize`, []int{DECOLORIZE}},
+		{`123`, []int{NUMBER}},
+		{`-123`, []int{SUB, NUMBER}},
```

Comment:
I reverted the changes in the lexer so all negative numbers now evaluate as SUB, NUMBER. The AST does the work now.